engine/src/ui/canvas/plugin.cpp

```diff
@@ -116,4 +134,58 @@ void cubos::engine::uiCanvasPlugin(Cubos& cubos)
                 window->renderDevice().clearDepth(1);
             }
         });
+
+    cubos.system("draw ui")
+        .tagged(uiEndTag)
+        .with<UIElement>()
+        .related<ChildOf>()
+        .with<UICanvas>()
+        .other()
+        .with<UIElement>()
+        .related<ChildOf>()
+        .with<UIElement>()
+        .call([](Query<Entity, UIElement&, const UICanvas&> drawsToQuery,
+                 Query<Entity, UIElement&, UIElement&> childrenQuery,
+                 Query<Entity, UICanvas&, RenderTarget&> canvasQuery, const core::io::Window& window) {
+            cubos::core::gl::DepthStencilStateDesc dssd;
+            dssd.depth.enabled = true;
+            dssd.depth.writeEnabled = true;
+            dssd.depth.near = -1.0F;
+            dssd.depth.compare = Compare::LEqual;
+            window->renderDevice().setDepthStencilState(window->renderDevice().createDepthStencilState(dssd));
+
+            for (auto [entity, canvas, renderTarget] : canvasQuery)
+            {
+                window->renderDevice().setViewport(0, 0, static_cast<int>(renderTarget.size.x),
+                                                   static_cast<int>(renderTarget.size.y));
+                auto mvpBuffer = window->renderDevice().createConstantBuffer(sizeof(glm::mat4), &canvas.mat,
+                                                                             cubos::core::gl::Usage::Static);
```

Comment:
Same as above, this should only be created once as it is an expensive operation. You can reuse it every frame.